### PR TITLE
Expose tests.cluster in ci-op's config

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2054

/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

File: pkg/api/domain.go [hunk 1]
```diff
@@ -3,11 +3,16 @@ package api
...
 )
 
 const (
-	ClusterNameAPICI = "api.ci"
-	ClusterNameAPPCI = "app.ci"
+	ClusterNameAPICI   = "api.ci"
+	ClusterNameAPPCI   = "app.ci"
+	ClusterNameBuild01 = "build01"
+	ClusterNameBuild02 = "build02"
+	ClusterNameVSphere = "vsphere"
```
Comment (petr-muller) [inline pkg/api/domain.go:15]: Cluster names seem to be unrelated to "domains", why include them in this file?

Comment (hongkailiu) [reply pkg/api/domain.go]: Moved them into `constant.go` in the same pkg.

File: pkg/api/domain.go [hunk 2]
```diff
@@ -18,6 +17,10 @@ const (
 	ServiceDomainAPPCIRegistry = "registry.ci.openshift.org"
 )
 
+var (
+	ValidClusterNames = sets.NewString(ClusterNameAPPCI, ClusterNameBuild01, ClusterNameBuild02, ClusterNameVSphere)
```
Comment (stevekuznetsov) [inline pkg/api/domain.go:21]: Why is this in a different file from the constants?

File: pkg/api/constant.go [hunk 3]
```diff
@@ -1,6 +1,12 @@
 package api
 
 const (
+	ClusterNameAPICI   = "api.ci"
```
Comment (stevekuznetsov) [inline pkg/api/constant.go:4]: Why do we have these _and_ the typed ones?

File: pkg/dispatcher/config.go [hunk 4]
```diff
@@ -41,19 +41,7 @@ type Config struct {
...
-	ClusterAPICI ClusterName = "api.ci"
-	// ClusterAPPCI is the cluster app.ci which runs Prow
-	ClusterAPPCI ClusterName = "app.ci"
-	// ClusterBuild01 is the cluster build01 in the build farm
-	ClusterBuild01 ClusterName = "build01"
-	// ClusterBuild02 is the cluster build02 in the build farm
-	ClusterBuild02 ClusterName = "build02"
-	ClusterVSphere ClusterName = "vsphere"
-)
+type ClusterName = api.Cluster
```
Comment (stevekuznetsov) [inline pkg/dispatcher/config.go:44]: Why? Just do the refactor

File: pkg/api/domain.go (hunk 2 above)
Comment (hongkailiu) [reply pkg/api/domain.go]: Moved to `constant.go`.

File: pkg/api/constant.go (hunk 3 above)
Comment (hongkailiu) [reply pkg/api/constant.go]: Removed the string ones.

File: pkg/dispatcher/config.go (hunk 4 above)
Comment (hongkailiu) [reply pkg/dispatcher/config.go]: Done.

**Review by AlexNPavel — APPROVED** [review]
LGTM
